src/EditorFeatures/CSharpTest/SplitStringLiteral/SplitStringLiteralCommandHandlerTests.cs

```diff
@@ -832,5 +836,53 @@ void M()
     }
 }");
         }
+
+        [WorkItem(40277, "https://github.com/dotnet/roslyn/issues/40277")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitStringLiteral)]
+        public void TestInStringWithKeepTabsEnabled1()
+        {
+            TestHandled(
+@"class C
+{
+	void M()
+	{
+		var s = ""Hello [||]world"";
+	}
+}",
+@"class C
+{
+	void M()
+	{
+		var s = ""Hello "" +
+			""[||]world"";
+	}
+}",
+            useTabs: true);
+        }
+
+        [WorkItem(40277, "https://github.com/dotnet/roslyn/issues/40277")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitStringLiteral)]
+        public void TestInStringWithKeepTabsEnabled2()
+        {
+            TestHandled(
+@"class C
+{
+	void M()
+	{
+		var s = ""Hello "" +
+			""there [||]world"";
+	}
+}",
+@"class C
+{
+	void M()
+	{
+		var s = ""Hello "" +
+			""there "" +
+			""[||]world"";
+	}
+}",
+            useTabs: true);
+        }
```

Comment:
for tests with tabs, i recommend having the test harness actually validate there are tabs in the before/after. to ensrue we don't accidentally lose this with an errant edit.